File: tests/expected/str_enum.kt


enum class Colors(
    val value: String,
) {
    RED("red"),
    GREEN("green"),
    BLUE("blue"),
}

fun show() {
    val color_map = hashMapOf(Colors.RED to "1", Colors.GREEN to "2", Colors.BLUE to "3")
    val a = Colors.GREEN
    if (a == Colors.GREEN) {
        println("green")
    } else {
        println("Not green")
    }
    if (true) {
        val __tmp1 = color_map.size
        println("$__tmp1")
    }
}

fun main(argv: Array<String>) {
    show()
}

File: tests/expected/str_enum.dart
// @dart=3.4
import 'package:sprintf/sprintf.dart';

enum Colors {
  RED("red"),
  GREEN("green"),
  BLUE("blue");

  const Colors(this.__private);
  final String __private;
}

show() {
  final Map<Colors, String> color_map = {
    Colors.RED: "1",
    Colors.GREEN: "2",
    Colors.BLUE: "3"
  };
  final Colors a = Colors.GREEN;

  if (a == Colors.GREEN) {
    print(sprintf("%s", ["green"]));
  } else {
    print(sprintf("%s", ["Not green"]));
  }
  print(sprintf("%s", [color_map.length]));
}

main(List<String> argv) {
  show();
}
